src/runtime/pkg/govmm/qemu/qemu.go

```diff
@@ -389,6 +387,14 @@ func (object Object) QemuParams(config *Config) []string {
 		// Add OVMF firmware as pflash drive
 		driveParams = append(driveParams, "if=pflash,format=raw,readonly=on")
 		driveParams = append(driveParams, fmt.Sprintf("file=%s", object.File))
+	case SNPGuest:
+		objectParams = append(objectParams, string(object.Type))
+		objectParams = append(objectParams, fmt.Sprintf("id=%s", object.ID))
+		objectParams = append(objectParams, fmt.Sprintf("cbitpos=%d", object.CBitPos))
+		objectParams = append(objectParams, fmt.Sprintf("reduced-phys-bits=%d", object.ReducedPhysBits))
+		// Add OVMF firmware as pflash drive
+		driveParams = append(driveParams, "if=pflash,format=raw,readonly=on")
+		driveParams = append(driveParams, fmt.Sprintf("file=%s", object.File))
```

Comment:
Indeed launching with policy=0 fails because (I think) bit 17 of the policy must be one (section 4.3 of SNP ABI spec):

![image](https://user-images.githubusercontent.com/9211/204733153-1c9a956c-2175-4bd6-9607-109086ceb241.png)

In this PR we remove `policy=%d` entirely, and therefore cause QEMU to use its default SNP policy value of 0x30000 (which means: bit17=1 and SMT=1 to allow SMT).

Don't we want to allow the policy to be configurable (like in SEV), but with the proper default for SNP?

Here's the QEMU patch that adds the SNP policy property (among other SNP properties): https://github.com/AMDESE/qemu/commit/05c0dd2c8ff2